scanners/ios/utility.go

```diff
@@ -615,101 +613,19 @@ func GenerateDefaultOptions(projectType XcodeProjectType) models.OptionNode {
-	xcodeArchiveMacStepInputModels := append(xcodeStepInputModels, envmanModels.EnvironmentItemModel{ExportMethodInputKey: "$" + ExportMethodEnvKey})
-
-	if hasTest {
-		switch projectType {
-		case XcodeProjectTypeIOS:
-			configBuilder.AppendStepListItemsTo(models.PrimaryWorkflowID, steps.XcodeTestStepListItem(xcodeStepInputModels...))
-		case XcodeProjectTypeMacOS:
-			configBuilder.AppendStepListItemsTo(models.PrimaryWorkflowID, steps.XcodeTestMacStepListItem(xcodeStepInputModels...))
-		}
-	} else {
-		switch projectType {
-		case XcodeProjectTypeIOS:
-			configBuilder.AppendStepListItemsTo(models.PrimaryWorkflowID, steps.XcodeArchiveStepListItem(xcodeArchiveStepInputModels...))
-
-			if shouldAppendExportAppClipStep(hasAppClip, exportMethod) {
-				appendExportAppClipStep(configBuilder, models.PrimaryWorkflowID)
-			}
-		case XcodeProjectTypeMacOS:
-			configBuilder.AppendStepListItemsTo(models.PrimaryWorkflowID, steps.XcodeArchiveMacStepListItem(xcodeArchiveMacStepInputModels...))
-		}
-	}
-
-	configBuilder.AppendStepListItemsTo(models.PrimaryWorkflowID, steps.DefaultDeployStepList(isIncludeCache)...)
-
-	if hasTest {
-		// CD
-		configBuilder.AppendStepListItemsTo(models.DeployWorkflowID, steps.DefaultPrepareStepList(isIncludeCache)...)
-		configBuilder.AppendStepListItemsTo(models.DeployWorkflowID, steps.CertificateAndProfileInstallerStepListItem())
-
-		if missingSharedSchemes {
-			configBuilder.AppendStepListItemsTo(models.DeployWorkflowID, steps.RecreateUserSchemesStepListItem(
-				envmanModels.EnvironmentItemModel{ProjectPathInputKey: "$" + ProjectPathInputEnvKey},
-			))
-		}
-
-		if hasPodfile {
-			configBuilder.AppendStepListItemsTo(models.DeployWorkflowID, steps.CocoapodsInstallStepListItem())
-		}
-
-		if carthageCommand != "" {
-			configBuilder.AppendStepListItemsTo(models.DeployWorkflowID, steps.CarthageStepListItem(
-				envmanModels.EnvironmentItemModel{CarthageCommandInputKey: carthageCommand},
-			))
-		}
-
-		switch projectType {
-		case XcodeProjectTypeIOS:
-			configBuilder.AppendStepListItemsTo(models.DeployWorkflowID, steps.XcodeTestStepListItem(xcodeStepInputModels...))
-			configBuilder.AppendStepListItemsTo(models.DeployWorkflowID, steps.XcodeArchiveStepListItem(xcodeArchiveStepInputModels...))
-
-			if shouldAppendExportAppClipStep(hasAppClip, exportMethod) {
-				appendExportAppClipStep(configBuilder, models.DeployWorkflowID)
-			}
-		case XcodeProjectTypeMacOS:
-			configBuilder.AppendStepListItemsTo(models.DeployWorkflowID, steps.XcodeTestMacStepListItem(xcodeStepInputModels...))
-			configBuilder.AppendStepListItemsTo(models.DeployWorkflowID, steps.XcodeArchiveMacStepListItem(xcodeArchiveMacStepInputModels...))
-		}
-
-		configBuilder.AppendStepListItemsTo(models.DeployWorkflowID, steps.DefaultDeployStepList(isIncludeCache)...)
-	}
+	createPrimaryWorkflow(projectType, configBuilder, isPrivateRepository, includeCache, missingSharedSchemes, hasTest, hasPodfile, carthageCommand)
```

Comment:
NOTE: it might be better to use a single struct param, instead of this much direct params